mlflow/tracking/default_experiment/databricks_notebook_experiment_provider.py

```diff
@@ -5,53 +5,39 @@
 from mlflow.utils import databricks_utils
 from mlflow.utils.mlflow_tags import (
     MLFLOW_EXPERIMENT_SOURCE_ID,
-    MLFLOW_EXPERIMENT_SOURCE_TYPE,
 )
 
 
 class DatabricksNotebookExperimentProvider(DefaultExperimentProvider):
-    def in_context(self):
-        return databricks_utils.is_in_databricks_notebook()
-
-    def get_experiment_id(self):
-        return databricks_utils.get_notebook_id()
-
-
-class DatabricksRepoNotebookExperimentProvider(DefaultExperimentProvider):
-    _resolved_repo_notebook_experiment_id = None
+    _resolved_notebook_experiment_id = None
 
     def in_context(self):
-        return databricks_utils.is_in_databricks_repo_notebook()
+        return databricks_utils.is_in_databricks_notebook()
 
     def get_experiment_id(self):
-        if DatabricksRepoNotebookExperimentProvider._resolved_repo_notebook_experiment_id:
-            return DatabricksRepoNotebookExperimentProvider._resolved_repo_notebook_experiment_id
+        if DatabricksNotebookExperimentProvider._resolved_notebook_experiment_id:
+            return DatabricksNotebookExperimentProvider._resolved_notebook_experiment_id
 
         source_notebook_id = databricks_utils.get_notebook_id()
         source_notebook_name = databricks_utils.get_notebook_path()
         tags = {
-            MLFLOW_EXPERIMENT_SOURCE_TYPE: "REPO_NOTEBOOK",
             MLFLOW_EXPERIMENT_SOURCE_ID: source_notebook_id,
         }
 
-        # With the presence of the above tags, the following is a get or create in which it will
+        # With the presence of the source id, the following is a get or create in which it will
         # return the corresponding experiment if one exists for the repo notebook.
-        # If no corresponding experiment exist, it will create a new one and return
-        # the newly created experiment ID.
+        # For non-repo notebooks, it will raise an exception and we will use source_notebook_id
         try:
             experiment_id = MlflowClient().create_experiment(source_notebook_name, None, tags)
         except MlflowException as e:
             if e.error_code == databricks_pb2.ErrorCode.Name(
                 databricks_pb2.INVALID_PARAMETER_VALUE
             ):
-                # If repo notebook experiment creation isn't enabled, fall back to
-                # using the notebook ID
+                # If determined that it is not a repo notebook
                 experiment_id = source_notebook_id
             else:
                 raise e
```

Comment:
@annzhang-db I thought we would get a `RESOURCE_ALREADY_EXISTS` exception if we call `create_experiment()` with a non-repo notebook path that already exists. Is that what the backend does?

If the backend does indeed return `RESOURCE_ALREADY_EXISTS`, I think that would break the current implementation of `DatabricksNotebookExperimentProvider` in this PR; have we tested this thoroughly?

Reply:
Yes, we get a `RESOURCE_ALREADY_EXISTS` exception if we call `create_experiment()` with a non-repo notebook path that already exists AND no sourceType/sourceId tags passed in. Since we are passing in the sourceId here, it will actually go into the previous case (in the backend PR, ill leave a comment there) and raise INVALID_PARAMETER_VALUE error for sourceId but no sourceType.

Reply:
It will call create_experiment() with a sourceId tag, which will not fail with `RESOURCE_ALREADY_EXISTS`. Only if create_experiment() is called without sourceId tag will it fail with `RESOURCE_ALREADY_EXISTS`.